tests/integration/blockchain/test_wallet_commands.py

```diff
@@ -384,9 +387,16 @@ async def test_sync_with_encryption_and_password_change(self):
         data = await daemon2.jsonrpc_sync_apply('password2')
         # sync_apply doesn't save password if encrypt-on-disk is False
         self.assertEqual(wallet2.encryption_password, None)
-        # need to use new password2 in sync_apply
-        with self.assertRaises(InvalidPasswordError):
-            await daemon.jsonrpc_sync_apply('password', data=data['data'], blocking=True)
+
+        # Need to use new password2 in sync_apply. Attempts with other passwords
+        # should fail consistently with InvalidPasswordError.
+        random = Random('password')
+        for i in range(200):
+            bad_guess = ''.join(random.choices(string.digits + string.ascii_letters + string.punctuation, k=40))
+            self.assertNotEqual(bad_guess, 'password2')
+            with self.assertRaises(InvalidPasswordError):
+                await daemon.jsonrpc_sync_apply(bad_guess, data=data['data'], blocking=True)
```

Comment:
Is there a way to make this more deterministic without looping 200 times?

Reply:
I tried setting `account.init_vectors` values to make it deterministic, but it doesn't work because these code paths (`pack`, `unpack`) use `better_aes_encrypt`:
https://github.com/lbryio/lbry-sdk/blob/a7d7efecc7aa8aa5ef666b16358ad1b2066e8eb1/lbry/crypto/crypt.py#L45

That function internally chooses a different `init_vector` every time. I tried digging into the history to understand why `init_vector` can be specified for `aes_encrypt`, but not for `better_aes_encrypt`. Search came up empty, as it's been that way for a long time.

Is there some advantage to `better_aes_encrypt` choosing a new init_vector? Like you can't tell if someone changed their password or not?

Reply:
Also, if we freeze the test in this way, there will be no further bugs of this kind discovered by giving bad passwords.
Randomness is what discovered the new error message here https://github.com/lbryio/lbry-sdk/pull/3667/commits/a8f799912b4115d49994fae8f631081c3a16456e.